core/src/kvs/tx.rs

```diff
@@ -2955,6 +2959,43 @@ impl Transaction {
 		Ok(None)
 	}
 
+	pub async fn define_in_out_fd_from_relation(
+		&mut self,
+		ns: &str,
+		db: &str,
+		tb: &str,
+		rel: &Relation,
+	) -> Result<(), Error> {
+		let in_kind = rel.from.clone().unwrap_or(Kind::Record(vec![]));
+		let out_kind = rel.to.clone().unwrap_or(Kind::Record(vec![]));
+
+		let in_key = crate::key::table::fd::new(ns, db, tb, "in");
+		let out_key = crate::key::table::fd::new(ns, db, tb, "out");
+
+		// TODO: fix permissions so they don't defalut to full
+		self.set(
+			in_key,
+			DefineFieldStatement {
+				name: Idiom(vec![Part::from("in")]),
+				what: tb.into(),
+				kind: Some(in_kind),
+				..Default::default()
+			},
+		)
+		.await?;
+		self.set(
+			out_key,
+			DefineFieldStatement {
+				name: Idiom(vec![Part::from("out")]),
+				what: tb.into(),
+				kind: Some(out_kind),
+				..Default::default()
+			},
+		)
+		.await?;
+		Ok(())
+	}
+
```

Comment:
Yeah this probably shouldn't go in this file. The only reason some of these functions go in this file, is because of their caching functionality. This doesn't have caching, so it shouldn't be here. Take a look at this PR:

https://github.com/surrealdb/surrealdb/pull/3146/files#diff-402ba7dc1d0e64c176bf293deda2320e98ff469e2f7e273c842fb278ab484910

Reply:
I'll have a look